sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/BatchSpannerRead.java

```diff
@@ -133,7 +130,7 @@ public void processElement(ProcessContext c) throws Exception {
     }
   }
 
-  private static class ReadFromPartitionFn extends DoFn<Partition, Struct> {
+  private static class ReadFromPartitionFn extends DoFn<List<Partition>, Struct> {
```

Comment:
You may want to mark it as `@BoundedPerElement`.

And are we able to provide better size information here?

Reply:
Added @BoundedPerElement.
About size information, we have a list of partitions to process, so we have the size of the total elements to process, which I think would be the best information to have about the size.